tests/e2e/cucumber/steps/app-files/resource.ts

```diff
@@ -144,6 +144,43 @@ When(
   }
 )
 
+Then(
+  /^"([^"]*)" (should|should not) be able to empty the trashbin?$/,
+  async function (this: World, stepUser: string, actionType: string): Promise<void> {
+    const { page } = this.actorsEnvironment.getActor({ key: stepUser })
+    const resourceObject = new objects.applicationFiles.Resource({ page })
+    const message = await resourceObject.emptyTrashBin()
+    if (actionType === 'should not') {
+      expect(message).toBe('failed to delete all files permanently')
+    } else {
+      expect(message).toBe('all deleted files were removed')
+    }
+  }
+)
+
+Then(
+  /^"([^"]*)" (should|should not) be able to delete following resource(s) from the trashbin?$/,
+  async function (
+    this: World,
+    stepUser: string,
+    actionType: string,
+    _: string,
+    stepTable: DataTable
+  ): Promise<void> {
+    const { page } = this.actorsEnvironment.getActor({ key: stepUser })
+    const resourceObject = new objects.applicationFiles.Resource({ page })
+    for (const info of stepTable.hashes()) {
+      const message = await resourceObject.deleteTrashBin({ resource: info.resource, actionType })
+      const paths = info.resource.split('/')
+      if (actionType === 'should not') {
+        expect(message).toBe(`failed to delete "${paths[paths.length - 1]}"`)
```

Comment:
you don't check that response returns 403 or 204.  instead that you check to message (which is also very good for e2e). 
How do you think, maybe also nice to have response checking?

Reply:
Checked here in the latest commit, if the status code is 204 or 403 and the assertion of the error message is done in the step definition. Does this look good to you?
https://github.com/owncloud/web/blob/a0ec2c47377fbc769653b0382bbf3870e4678edb/tests/e2e/support/objects/app-files/resource/actions.ts#L376-L383

Reply:
looks good, thank you